stock-filters/ChirpVillage/Builder.py

```diff
@@ -0,0 +1,257 @@
+import utilityFunctions
+import RandUtils
+from Biomes import BlockUtils
+
+
+"""Utils for buildings"""
+
+def get_coords(coords): 
+    return coords[0][0], coords[0][1], coords[0][2], coords[1][0], coords[1][1]
+
+def construct_walls(level, coords, biome, height):
+    minx, minz, miny, maxx, maxz = get_coords(coords)
+    wall_block = BlockUtils.get_wall_block(biome)
+    # walls
+    for y in range(height, miny, -1):
+        for x in range(minx, maxx):
+            for z in range(minz, maxz):
+                # Skip columns since they should be a different block
+                if (x == minx and z == minz) or (x == maxx-1 and z == minz) or \
+                        (x == minx and z == maxz-1) or (x == maxx-1 and z == maxz-1):
+                    continue
+                if x == maxx-1 or z == maxz-1 or x == minx or z == minz:
+                    utilityFunctions.setBlock(
+                        level, wall_block, x, y, z)
+
+def construct_floor_and_flat_roof(level, coords, biome, height):
+    minx, minz, miny, maxx, maxz = get_coords(coords)
+    roof_block = BlockUtils.get_beam_block(biome)
+    floor_block = BlockUtils.get_floor_block(biome)
+
+    # roof and floor
+    for x in range(minx, maxx):
+        for z in range(minz, maxz):
+            utilityFunctions.setBlock(
+                level, roof_block, x, height, z)
+            utilityFunctions.setBlock(
+                level, floor_block, x, miny, z)
+
+def construct_pointed_roof(level, coords, biome, height):
+    minx, minz, miny, maxx, maxz = get_coords(coords)
+    roof_block = BlockUtils.get_beam_block(biome)
+    for i in range(-1, maxx-minx/4):
+        for x in range(minx+i, maxx-i):
+            for z in range(minz+i, maxz-i):
+                utilityFunctions.setBlock(
+                    level, roof_block, x, height+i, z)
+                
+def construct_pillars(level, coords, biome, height):
+    minx, minz, miny, maxx, maxz = get_coords(coords)
+    pillar_block = BlockUtils.get_beam_block(biome)
+    for y in range(miny, height):
+        for x in [minx, maxx-1]:
+            for z in [minz, maxz-1]:
+                utilityFunctions.setBlock(
+                    level, pillar_block, x, y, z)
+                
+def place_door(level, coords, biome):
+    minx, minz, miny, maxx, maxz = get_coords(coords)
+    door_block = BlockUtils.get_door_block(biome)
+    # Place door
+    utilityFunctions.setBlock(
+        level, door_block, (minx+maxx)/2, miny+1, minz)
```

Comment:
you can get the door location from the yard generator now

Reply:
This code currently uses the entire bounding box anyway, The various building coords are not passed in. 

Integration of this code with the yards can come in another PR